src/foam/nanos/ruler/RulerDAO.js

```diff
@@ -123,14 +119,77 @@
         }
       ],
       javaCode: `
-
       for ( Object key : sink.getGroupKeys() ) {
         List<Rule> group = ((ArraySink) sink.getGroups().get(key)).getArray();
         if ( ! group.isEmpty() ) {
-          new RuleEngine(x, this).execute(group, obj, oldObj);
+          new RuleEngine(x, getDelegate()).execute(group, obj, oldObj);
```

Comment:
Use RulerDAO as the delegate of rule engine so that RulerDAO.cmd_ would be called https://github.com/foam-framework/foam2/pull/1947/commits/7edd451b5f8c20e4ad485d5648023aad9f4098a2.